core/src/test/scala/integration/kafka/api/AdminClientTest.scala

```diff
@@ -76,6 +90,15 @@ class AdminClientTest extends IntegrationTestHarness with Logging {
     assertEquals("consumer", group.protocolType)
   }
 
+  @Test(expected = classOf[TimeoutException])
+  def testListGroupsTimeout() {
+    val map = client.listAllGroups()
```

Comment:
A mocked instance of Node wouldn't offer much. The Node class purely contains information to reach the server. This information is checked for validity so an instance of noNode also could not be used.